### PR TITLE
Bot API 5.1: Support Improved Invite Links and new messageTypes

### PR DESCRIPTION
**Pull request by danielperez9430**
- [X] All tests pass
- [X] I have run `npm run doc`

### Description

- Added the method createChatInviteLink
- Added the method editChatInviteLink
- Added the method revokeChatInviteLink
- Add support for new messageTypes:
     * *voice_chat_started*
     * *voice_chat_ended*
     * *voice_chat_participants_invited*
     * *message_auto_delete_timer_changed*
     * *chat_invite_link*
     * *chat_member_updated*
- Add support for new updates:
     * *my_chat_member*
     * *chat_member*

### References

* Telegram Bot API - https://core.telegram.org/bots/api-changelog#march-9-2021

### CODE REVIEW

**Comment by danielperez9430** [comment]
## Example

```js
bot.onText(/link/, (msg) => {

  var chat_id = msg.chat.id;


  var date = (new Date(2021, 03, 17)).getTime(); // Timestamp date

  // Options
  var ops = {
    expire_date: date,
    member_limit: 1
  }

  bot.createChatInviteLink(chat_id, ops).then((data) => {
    console.log(data);

    bot.editChatInviteLink(chat_id, data.invite_link, { member_limit: 3 }).then(data2 => {
      console.log(data2);
      
      bot.revokeChatInviteLink(chat_id, data2.invite_link).then(data3 => {
        console.log(data3);
      });
    });


  });

});
```

**Review by snatvb — APPROVED** [review]

**Comment by snatvb** [comment]
Hi, @danielperez9430 , Good work! I builded your fork and it works. Thank you. I hope that it will be merged as soon.

**Comment by danielperez9430** [comment]
In the afternoon I add the missing tests and update version

**Comment by danielperez9430** [comment]
imagen

All test OK

**Comment by danielperez9430** [comment]
Example new MessageTypes:

```js
bot.on('voice_chat_started', (data) => {
  console.log(data);
});

bot.on('voice_chat_ended', (data) => {
  console.log(data);
});

bot.on('voice_chat_participants_invited', (data) => {
  console.log(data);
});

bot.on('message_auto_delete_timer_changed', (data) => {
  console.log(data);
});
```

File: doc/api.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
-# API Reference
```
Comment (kamikazechaser) [inline doc/api.md:1]: These changes will break the docs. Are you using Windows by any chance?

**Review by kamikazechaser — CHANGES_REQUESTED** [review]
The following update types have been added too:
- my_chat_member/chat_member
- chat_member

New types:
- chat_invite_link (This would be useful for administration bots)
- chat_member_updated

File: doc/api.md (hunk 1 above)
Comment (danielperez9430) [reply doc/api.md:1]: Yes, I use Windows. But I can do the npm run doc from Linux if need be

**Comment by danielperez9430** [comment]
> The following update types have been added too:
> - my_chat_member/chat_member
> - chat_member
> 
> New types:
> - chat_invite_link (This would be useful for administration bots)
> - chat_member_updated

I'll check it this afternoon and send you a message on Telegram

**Comment by danielperez9430** [comment]
Example:

```js
bot.on('my_chat_member', data => {
  console.log(data);
});
```

Note: The following event occurs when a user blocks / unblocks the bot, changes its permissions in a group, ...

**Review by kamikazechaser — APPROVED** [review]

**Pull request merged**